### PR TITLE
feat(displays): Add pixelated screens

### PR DESCRIPTION
**Pull request by bouveng**
<p align="right" style="font-size:10px;"><i>Screens with different pixel density. DCDU checkered pixel pattern.<br/>(watch images at 100% to see effect properly)</i></p>

## Summary of Changes
:closed_book::green_book::blue_book:

This PR adds **pixels** to the following screens:

- `PFD` - pixels, tint and subtle backlight bleed
- `ND` - pixels, tint and subtle backlight bleed
- `EICAS 1 + 2` - pixels, tint and subtle backlight bleed
- `DCDU 1 + 2` - pixels (checkered), tint
- `SAI/ISIS` - pixels, tint
- `FCU/MCDU` - pixels, tint

Large screens have a subtle blue tint and a small backlight bleed that can be seen in the dark in a favorable setting (dark and high intensity setting).
DCDU screens have a light white tint and a checkered large pixel pattern.

## Screenshots

<p align="right"><i>MCDU, SAI/ISIS, letters and WX.<br/>(watch images at 100% to see effect properly)</i></p>

Microsoft Flight Simulator 2021-07-23 23_35_53
<p align="right"><i>Subtle small backlight bleed can be seen on the large screens in favorable settings (dark and high intensity setting)<br/>(watch image at 100% to see effect properly)</i></p>

Microsoft Flight Simulator 2021-07-24 00_00_30
<p align="right"><i>Dust and scratches are still visible.<br/>(watch image at 100% to see effect properly)</i></p>

## References

<p align="right"><i>Note the DCDU's checkered pixel pattern</i></p>

## Known behaviors
At some zoom levels and angles, a very minor unavoidable [moiré effect](https://en.wikipedia.org/wiki/Moir%C3%A9_pattern) can be observed.

Discord username: @bouveng

## Testing instructions

1, Spawn on runway
2, Observe screens at different zoom levels and under various lighting conditions.<br/>(_you need to be somewhat close to see the pixels at all_)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/CSS/A32NX_Display_Common.css [hunk 1]
```diff
@@ -46,6 +46,28 @@ text {
...
+  width: 100%;
+  height: 100%;
+  z-index: 999;
+  position: absolute;
+  /* custom svg */
+  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 10 10'%3E%3Cpath d='M0 0 V10 Z H10 Z' stroke='%23000' stroke-width='3' fill='none'/%3E%3C/svg%3E");
+  background-size: 2px;
+  opacity: 0.75;
+  /* subtle tint and edge bleed */
+  background-color:rgba(0, 0, 255, 0.025);
```
Comment (hiaaryan) [inline flybywire-aircraft-a320-neo/html_ui/CSS/A32NX_Display_Common.css:61]: `background-color: rgba(0, 0, 255, 0.025);`

File: src/instruments/src/Common/pixels.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  @if $checkered {
+    background-image: url("data:image/svg+xml,%3Csvg width='2' height='2' xmlns='http://www.w3.org/2000/svg'%3E%3Crect height='1' width='1' y='0' x='0' /%3E%3Crect height='1' width='1' y='1' x='1' /%3E%3C/svg%3E");
+    background-position: 1px 0;
+    background-color: rgba(80, 80, 80, 0.2);
+  } @else {
+    background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 10 10'%3E%3Cpath d='M0 0 V10 Z H10 Z' stroke='%23000' stroke-width='#{$spacing}' fill='none'/%3E%3C/svg%3E");
+    @if $bleed {
+      /* subtle tint, bleed and edge bleed */
+      background-color:rgba(0, 0, 255, 0.025);
```
Comment (hiaaryan) [inline src/instruments/src/Common/pixels.scss:19]: `background-color: rgba(0, 0, 255, 0.025);`

File: flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml [hunk 3]
```diff
@@ -513,6 +501,15 @@
                     <POTENTIOMETER>88</POTENTIOMETER>
                 </UseTemplate>
 
+                <UseTemplate Name="ASOBO_GT_Component_Emissive_Gauge">
+                    <NODE_ID>SCREEN_PFD_L</NODE_ID>
+                    <POTENTIOMETER>88</POTENTIOMETER>
+                    <PART_ID>SCREEN_PFD1</PART_ID>
+                    <CAMERA_TITLE>PFD</CAMERA_TITLE>
+                    <!-- Pixelated screens needs more light -->
+                    <EMISSIVE_CODE>2</EMISSIVE_CODE>
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml:510]: Doesn't `EMISSIVE_CODE` break the potentiometers ?

Comment (bouveng) [reply flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml:510]: No. Potentiometers works fine. This value acts as a "`SET_CODE`", sort of, and this value gets multiplied on top of the value that controls the emissive. If you walk up the template tree you see where I end up.

**Review by Benjozork — APPROVED** [review]

**Comment by NUTTTTTTTT** [comment]
Quality Assurance Report

Discord: █▀█ █▄█ ▀█▀#2123
Object of testing: #5385
Tier of Testing: 1?
Date: (24/07/2021)

Testing Process: Spawned on runway, tested at varying times of day. Tested at different brightnesses and zoomed in on the various displays from different angles to check the screens.


Negatives: Absolutely N/A
Testing Results: Passed 

Conclusions: Looks brilliant under all light conditions, pixels are barely visible from a distance and adds a pretty nice immersion. No issues at different screen brightness'. 

All in all, a brilliant addition, kudos!
image
image

**Review by hiaaryan — APPROVED** [review]

**Review by davidwalschots — APPROVED** [review]
Awesome!

File: flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml (hunk 3 above)
Comment (bouveng) [reply flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml:510]: @Benjozork Just for reference, in addition to the above:
```
<UseTemplate Name="ASOBO_GT_Emissive_Gauge">
    <EMISSIVE_CODE>(A:LIGHT POTENTIOMETER:#POTENTIOMETER#, Percent over 100) #EMISSIVE_CODE# *</EMISSIVE_CODE>
</UseTemplate>
```
This is where that snippet ends up in the end.

**Comment by TheSledge-SLEJ** [comment]
Beautiful x

**Pull request merged**